### PR TITLE
feat: added dynamic blog ui

### LINKED ISSUE #6: Build a dynamic blog page UI
build out a dynamic blog page UI for peer protocol that would take data from an array of text objects. Take your design inspo from https://dribbble.com/shots/24565936-Minimal-blog-Untitled-UI and when you are done, the url to posts on the blog page should be shown as :

https://peer-on-stark.vercel.app/blog/{name of the blog post}

### PR DESCRIPTION
**Pull request by stephanniegb**
resolves #6 

## Release Notes

- **New Features**
  - Introduced a new blog component that displays individual blog posts with error handling for invalid IDs.
  - Added a layout component for consistent navigation and footer across blog pages.
  - Launched a featured blog section along with a grid of additional blog entries.

- **Data Structure**
  - Added a JSON file containing structured data for multiple blog entries.
  - Introduced a JSON schema to define the structure of blog data.

- **Styling Updates**
  - Updated global CSS variables for improved color management using hexadecimal values.

- **Type Definitions**
  - Introduced new TypeScript types for enhanced structure and type safety in blog data handling.

### CODE REVIEW

File: frontend/src/app/blog/[name]/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import Image from "next/image";
+import { useParams } from "next/navigation";
+import BlogsDataJson from "../../../../data/blogs.json";
+import { BlogsDataType } from "@/types";
+
+const Blog = () => {
+  const BlogsData: BlogsDataType = BlogsDataJson;
+  const params = useParams();
+  const blogId = params.id as string;
+  const blogData = BlogsData.entries[blogId];
```
Comment (Dprof-in-tech) [inline frontend/src/app/blog/[name]/[id]/page.tsx:11]: add an optional check for the blogid if its not available

File: frontend/src/app/blog/[name]/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        <h1 className="flex items-center gap-2">
+          <span className="text-3xl">404 |</span>
+          <span>This page could not be found.</span>
+        </h1>
+      </div>
+    );
+  }
+
+  return (
+    <section className="p-2">
```
Comment (Dprof-in-tech) [inline frontend/src/app/blog/[name]/[id]/page.tsx:25]: You are not using the theme from the theme provider so the page is perpetually in dark mode. please fix it

File: frontend/src/app/blog/[name]/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  text,
+  sectionId,
+}: BlogSectionProp) => {
+  return (
+    <div id={sectionId} className="flex flex-col gap-8">
+      {heading && <h2 className="text-3xl capitalize">{heading}</h2>}
+      {img && (
+        <div className="relative w-full max-w-[500px] h-[350px]">
+          <Image src={img} alt={img_alt} className="rounded-[4px]" fill />
+        </div>
```
Comment (Dprof-in-tech) [inline frontend/src/app/blog/[name]/[id]/page.tsx:120]: the images do not show up on the preview deployment, could you fix that?

File: frontend/yarn.lock [hunk 4]
```diff
@@ -2862,6 +2862,11 @@ fs.realpath@^1.0.0:
   resolved "https://registry.npmjs.org/fs.realpath/-/fs.realpath-1.0.0.tgz"
```
Comment (Dprof-in-tech) [inline frontend/yarn.lock:2862]: please remove frontend/yarn.lock from the commit. its an auto generated file

**Review by Dprof-in-tech — CHANGES_REQUESTED** [review]

File: frontend/src/app/blog/[name]/[id]/page.tsx (hunk 1 above)
Comment (stephanniegb) [reply frontend/src/app/blog/[name]/[id]/page.tsx:11]: There is a check on line 13 for the blog id, is that what you mean?

File: frontend/src/app/blog/[name]/[id]/page.tsx (hunk 2 above)
Comment (stephanniegb) [reply frontend/src/app/blog/[name]/[id]/page.tsx:25]: done

File: frontend/src/app/blog/[name]/[id]/page.tsx (hunk 3 above)
Comment (stephanniegb) [reply frontend/src/app/blog/[name]/[id]/page.tsx:119]: worked on it, does it show up now?

File: frontend/yarn.lock (hunk 4 above)
Comment (stephanniegb) [reply frontend/yarn.lock]: k

File: frontend/src/app/blog/[name]/[id]/page.tsx (hunk 1 above)
Comment (Dprof-in-tech) [reply frontend/src/app/blog/[name]/[id]/page.tsx:11]: thats okay

File: frontend/src/app/blog/[name]/[id]/page.tsx (hunk 3 above)
Comment (Dprof-in-tech) [reply frontend/src/app/blog/[name]/[id]/page.tsx:112]: yes it does now.

**Pull request merged**